enhancements/windows-containers/monitoring-windows-nodes.md

```diff
@@ -0,0 +1,202 @@
+  created in the namespace consist of subsets of endpoints from all the
+  Windows nodes.
+* Create a Service Monitor in `openshift-windows-machine-config-operator`
+  namespace for Service created above.
+  
+To display node graphs WMCO has done the following:
+
+* Add custom Prometheus rules in `openshift-windows-machine-config-operator`
+  namespace. The custom recording rules are created using Windows metrics
+  exposed by the windows_exporter and have the same names as Linux
+  recording rules. This is to make use of same console queries as Linux.
+* Note that WMCO is unable to display pod graphs for the Windows Nodes
+  with the current implementation. See [Risks and Mitigations](#risks-and-mitigations)
+  for details.
+  
+### Future Plans
+
+* As we move forward, our plan to display monitoring graphs is to create a
+ [common interface](https://issues.redhat.com/browse/WINC-530) for Windows
+  and Linux recording rules. Monitoring team will define recording rules for the
+  metrics that have different `metric labels` for Linux and Windows. The
+  differences in `metric labels` is captured in this [doc](https://docs.google.com/document/d/1-So18TTztD3ac3S9_8PigVkWcbatJC3IHAU5dOeWljc/edit#)
+  The Windows team will align the Windows recording rules with these new
+  recording rules. The recording rules for Windows will be managed by
+  WMCO. This set of common recording rules for monitoring will return results
+  for both Linux and Windows nodes for a single query.The console queries
+  currently use some raw metrics such as `node_filesystem_size_bytes`,
+  `node_filesystem_free_bytes` etc. They would need to be updated to include
+  the new recording rules in place of using raw metrics. This will ensure that
+  we have a consistent user experience for monitoring across Linux and Windows.
+* In the cases where `metric labels` are equivalent, we plan to relabel the
+  Windows metrics to align with the Linux metrics.  
+* Since the metrics Endpoints object is managed by WMCO, we plan to replace
+  Endpoints object with [EndpointSlices](https://kubernetes.io/docs/concepts/services-networking/endpoint-slices/#motivation)
+  to improve performance. This can be done once the `prometheus-operator` has
+  [support](https://github.com/prometheus-operator/prometheus-operator/issues/3862)
+  for EndpointSlices object.
+* Since the windows-exporter is not running as a [pod](#justification), the
+  endpoint is not secure. The reason for this is when running inside a pod, we
+  can use CA signer for providing TLS cert/key to the service for
+  authentication. We plan to leverage windows_exporter's support for `https`
+  configuration. WMCO will be responsible for adding [web config](https://github.com/prometheus/exporter-toolkit/blob/master/docs/web-configuration.md)
+  for TLS. This will ensure that the metrics Endpoint will be able to
+  authenticate the requests.
+* We plan to ensure that for telemetry rules also use metrics from Windows. This
```

Comment:
Linking to the existing telemetry rules will help during implementation of the KEP.